libsolidity/formal/SMTEncoder.cpp

```diff
@@ -666,7 +666,6 @@ void SMTEncoder::visitAssert(FunctionCall const& _funCall)
 	auto const& args = _funCall.arguments();
 	solAssert(args.size() == 1, "");
 	solAssert(args.front()->annotation().type->category() == Type::Category::Bool, "");
-	addPathImpliedExpression(expr(*args.front()));
```

Comment:
We don't assert it anymore to still reach further failing assertions.

Reply:
I still need to check what this did and understand why it's not done anymore.

Reply:
This can maybe be a separate PR.
What it does is basically follow the same as the generated EVM code: if the execution goes through the assertion, the condition must be true.
The issue is that if an assertion fails, an assertion that fails after the first failed one won't be reported, that's why I removed it.
It then doesn't follow the EVM semantics anymore, but I think that's fine since of course we check the assertion and report failures.

Reply:
Hm... if I understand that correctly, then I guess one can argue for both ways of doing things... but I'd need to look into it a bit more. Or maybe you can tell me: in the following two cases:
```
  function f(uint256 a, uint256 g) public { assert(a == 42); assert (b == 42); assert(a == b); }
```
What exactly would happen with and without this? If I understood correctly that'd be a good example... and the first two would always be reported as failing? And whether the third is reported or not is contingent on this change?

Reply:
Yes, that's what I expect too.
Another example would be something like
```
function f(uint x) {
  x = 12;
  assert(x == 13); // this creates a contradiction in the constraints local to this block
  // therefore any assertion from here on won't be reported
```

Reply:
But I agree with you. One can argue both ways

Reply:
Right, I hadn't thought about this easily introducing trivial contradictions - before that I would have thought the old way was better, but that may in fact be a good argument for preferring the new way... although you could still argue that it's fine not to report anything, if the code is unreachable anyways... so yeah - I'm satisfied with understanding what this does and I'm fine either way as well :-).